src/main/java/org/logstash/beats/Server.java

```diff
@@ -121,15 +118,11 @@ public void initChannel(SocketChannel socket) throws IOException, NoSuchAlgorith
                 SslHandler sslHandler = sslBuilder.build(socket.alloc());
                 pipeline.addLast(SSL_HANDLER, sslHandler);
             }
-
-            // We have set a specific executor for the idle check, because the `beatsHandler` can be
-            // blocked on the queue, this the idleStateHandler manage the `KeepAlive` signal.
-            pipeline.addLast(idleExecutorGroup, KEEP_ALIVE_HANDLER, new IdleStateHandler(0, IDLESTATE_WRITER_IDLE_TIME_SECONDS , clientInactivityTimeoutSeconds));
-
-            pipeline.addLast(BEATS_PARSER, new BeatsParser());
+            pipeline.addLast(idleExecutorGroup, IDLESTATE_HANDLER, new IdleStateHandler(clientInactivityTimeoutSeconds, IDLESTATE_WRITER_IDLE_TIME_SECONDS , clientInactivityTimeoutSeconds));
             pipeline.addLast(BEATS_ACKER, new AckEncoder());
-            pipeline.addLast(BEATS_HANDLER, new BeatsHandler(this.message));
-
+            pipeline.addLast(KEEP_ALIVE_HANDLER, new KeepAliveHandler());
+            pipeline.addLast(BEATS_PARSER, new BeatsParser());
+            pipeline.addLast(new DefaultEventExecutorGroup(DEFAULT_BEATS_HANDLER_THREAD), BEATS_HANDLER, new BeatsHandler(this.message));
```

Comment:
Need to create the ExecutorGroup in the same way that you created the idleExecutorGroup, otherwise each new connection will create an executor group